src/arrays/programs_involving_data_sequences/hard/q10/MainTest.java

```diff
@@ -0,0 +1,54 @@
+package arrays.programs_involving_data_sequences.hard.q10;
+
+import global.BaseTest;
+import global.MethodTest;
+import global.tools.CustomAssertions;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+public class MainTest extends BaseTest {
+    //Java
+    @Override
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("The middle letters of the array are\\: "),
+                new NewLine(),
+                new StringLiteral("delilah"),
+                new NewLine()
+        };
+    }
+
+    @Override
+    public void runMain() {
+        ArrayThereDelilah.main(new String[0]);
+    }
+
+    static Stream<Arguments> inputProvider() {
+        return Stream.of(
+                Arguments.of(new String[]{"add", "here", "lilly", "president", "jelly", "advance", "archers"}, "delilah".toCharArray()),
+                Arguments.of(new String[]{"archers", "get", "hello", "island", "cloud"}, "hello".toCharArray()),
+                Arguments.of(new String[]{"fireworks", "acorn", "array", "pilot", "hedge"}, "world".toCharArray()),
```

Comment:
You can also take a look at arrayUtil for this question. There is a method that will generate an array of random words for you, which would be useful for this method. However, if you did that you would have to create a helper method, but that should be fine as long as you don't look at the original code.

Reply:
I made that change but also kept the hardcoded ones.